src/Common/EventListener/ResponseSecurer.php

```diff
@@ -14,16 +14,38 @@ class ResponseSecurer
      */
     public function onKernelResponse(FilterResponseEvent $event): void
     {
+        $cacheControl = 'max-age=0, must-revalidate, private, no-cache';
+        if (defined('APPLICATION') && APPLICATION === 'Backend') {
+            $cacheControl .= ', no-store';
+        }
+
         $headers = [
+            'Cache-Control' => $cacheControl,
             'X-Frame-Options' => 'deny',
+            'Expires' =>  '0',
+            'Pragma' => 'no-cache',
             'referrer' => 'strict-origin-when-cross-origin',
+            'Referrer-Policy' => 'strict-origin-when-cross-origin',
             'X-XSS-Protection' => '1; mode=block',
             'X-Content-Type-Options' => 'nosniff',
         ];
 
+
+        $responseHeaders = $event->getResponse()->headers;
         foreach ($headers as $header => $value) {
-            if (!$event->getResponse()->headers->has($header)) {
-                $event->getResponse()->headers->set($header, $value);
+            if (!$responseHeaders->has($header)) {
+                $responseHeaders->set($header, $value);
+            }
+        }
+
+        // Don't leak server config
+        $blockedHeaders = [
+            'x-powered-by',
+            'Server',
+        ];
+        foreach ($blockedHeaders as $blockedHeader) {
+            if ($responseHeaders->has($blockedHeader)) {
+                $responseHeaders->remove($blockedHeader);
```

Comment:
Q: Do we need to check in a case-insensitive way? 
I think this header is often used in Nginx with `X-Powered-By` (camelcased version) next to the lowercase version.

Reply:
A: the keys are lowercased

```php
    /**
     * Returns true if the HTTP header is defined.
     *
     * @param string $key The HTTP header
     *
     * @return bool true if the parameter exists, false otherwise
     */
    public function has($key)
    {
        return \array_key_exists(str_replace('_', '-', strtolower($key)), $this->all());
    }
```
and

```php
/**
     * {@inheritdoc}
     */
    public function remove($key)
    {
        $uniqueKey = str_replace('_', '-', strtolower($key));
        unset($this->headerNames[$uniqueKey]);

        if ('set-cookie' === $uniqueKey) {
            $this->cookies = [];

            return;
        }

        parent::remove($key);

        if ('cache-control' === $uniqueKey) {
            $this->computedCacheControl = [];
        }

        if ('date' === $uniqueKey) {
            $this->initDate();
        }
    }
```

Reply:
thx, didn't know!